lib/pbench/server/api/__init__.py

```diff
@@ -48,59 +49,16 @@ def register_endpoints(api, app):
     api.add_resource(GraphQL, f"{app.config['REST_URI']}/graphql")
 
 
-def create_app():
+def create_app(server_config=ProdServerConfig):
     """Create Flask app with defined resource endpoints."""
 
     global app
 
-    cfg_name = os.environ.get("_PBENCH_SERVER_CONFIG")
-    if not cfg_name:
-        print(
-            f"{__name__}: ERROR: No config file specified; set"
-            " _PBENCH_SERVER_CONFIG",
-            file=sys.stderr,
-        )
-        sys.exit(1)
-
-    try:
-        config = PbenchServerConfig(cfg_name)
-    except BadConfig as e:
-        print(f"{__name__}: {e} (config file {cfg_name})", file=sys.stderr)
-        sys.exit(1)
-
     app = Flask(__name__)
     api = Api(app)
 
-    app.logger = get_pbench_logger(__name__, config)
-    app.config_server = config.conf["pbench-server"]
-    app.config_elasticsearch = config.conf["elasticsearch"]
-    app.config_graphql = config.conf["graphql"]
-
-    prdp = app.config_server.get("pbench-receive-dir-prefix")
-    if not prdp:
-        app.logger.error("Missing config variable for pbench-receive-dir-prefix")
-        sys.exit(1)
-    try:
-        upload_directory = Path(f"{prdp}-002").resolve(strict=True)
-    except FileNotFoundError:
-        app.logger.exception("pbench-receive-dir-prefix does not exist on the host")
-        sys.exit(1)
-    except Exception:
-        app.logger.exception(
-            "Exception occurred during setting up the upload directory on the host"
-        )
-        sys.exit(1)
-    else:
-        app.upload_directory = upload_directory
-
-    app.config["PORT"] = app.config_server.get("rest_port")
-    app.config["VERSION"] = app.config_server.get("rest_version")
-    app.config["MAX_CONTENT_LENGTH"] = filesize_bytes(
-        app.config_server.get("rest_max_content_length")
-    )
-    app.config["REST_URI"] = app.config_server.get("rest_uri")
-    app.config["BIND_HOST"] = app.config_server.get("bind_host")
-    app.config["WORKERS"] = app.config_server.get("workers")
+    app.config.from_object(server_config())
```

Comment:
`server_config()` here is instantiating an object of your class; I just think that constructor should be called from the client, passing in the object rather than a reference to the class.